File: jan1/jan1.rb
def two_sum(nums, target)
    num_map = {}
    nums.each_with_index do |num, idx|
        complement = target - num
        return [num_map[complement], idx] if num_map[complement]
        num_map[num] = idx
    end
    []
end

nums = [2, 7, 11, 15]
target = 9
result = two_sum(nums, target)
puts result

File: jan1/jan1.cs
using System;
using System.Collections.Generic;

class Program {
    public static int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> numMap = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            int complement = target - nums[i];
            if (numMap.ContainsKey(complement)) {
                return new int[] { numMap[complement], i };
            }
            numMap[nums[i]] = i;
        }
        return new int[] { };
    }

    static void Main() {
        int[] nums = { 2, 7, 11, 15 };
        int target = 9;
        int[] result = TwoSum(nums, target);
        foreach (int i in result) {
            Console.Write(i + " ");
        }
    }
}
